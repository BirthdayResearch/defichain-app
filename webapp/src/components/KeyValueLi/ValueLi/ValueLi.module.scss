// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../../scss/variables';
@import '../../../scss/bootstrap/variables';

.keyValueLi {
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  padding: $table-cell-padding-y 0;
  width: 100%;
}

.label {
  padding-right: 1rem;
}

.value {
  position: relative;
  display: grid;
  grid-template-columns: 230px 1fr;
  width: inherit;

  svg {
    display: block;
    color: $secondary;
  }
}

.copiedIndicator {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: $dfi-scent;
  align-items: center;
  justify-content: center;
  display: none;
}

.card {
  .copiedIndicator {
    background-color: $body-bg;
  }
}

.qr {
  display: block;
  margin-top: 4px;
  margin-bottom: 4px;
  padding: 8px;
}
