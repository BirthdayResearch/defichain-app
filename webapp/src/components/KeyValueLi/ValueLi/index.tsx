import React, { useState } from 'react';
import EllipsisText from 'react-ellipsis-text';
import CopyToClipboard from '../../CopyToClipboard';
import classnames from 'classnames';
import { I18n } from 'react-redux-i18n';
import styles from './ValueLi.module.scss';
import { onViewOnChain } from '../../../utils/utility';

interface KeyValueLiProps {
  copyable?: boolean | string;
  value: string;
  popsQR?: any;
  uid?: any;
  label?: string;
  textLimit?: number | string;
}

const ValueLi: React.FunctionComponent<KeyValueLiProps> = (
  props: KeyValueLiProps
) => {
  const [copied, changeCopied] = useState(false);

  const handleCopy = () => {
    changeCopied(true);
    setTimeout(() => {
      changeCopied(false);
    }, 600);
  };

  return (
    <>
      <a
        href='#'
        onClick={(e) => {
          e.preventDefault();
          e.stopPropagation();
          onViewOnChain(props.value);
        }}
      >
        <div className={styles.keyValueLi}>
          <div className={styles.value}>
            <div>
              <div
                className={classnames(
                  { 'd-flex': copied },
                  styles.copiedIndicator
                )}
              >
                {I18n.t('components.keyValueLi.copied')}
              </div>
              {props.value ? (
                <EllipsisText
                  text={props.value}
                  length={Number(props.textLimit ?? `50`)}
                />
              ) : (
                '-'
              )}
            </div>
          </div>
        </div>
      </a>
      <div>
        {props.value && props.copyable && (
          <CopyToClipboard
            value={props.value!}
            link={'default'}
            handleCopy={handleCopy}
          />
        )}
      </div>
    </>
  );
};

export default ValueLi;
