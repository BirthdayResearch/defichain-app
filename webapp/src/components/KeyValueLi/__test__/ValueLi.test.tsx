import React from 'react';
import ValueLi from '../ValueLi/index';
import { shallow } from 'enzyme';

describe('ValueLi component', () => {
  it('should check for snapshot', () => {
    const props = {
      copyable: '',
      value: 'value',
      popsQR: 'popsQR',
      uid: 'uid',
      label: 'label',
    };

    const wrapper = shallow(<ValueLi {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
