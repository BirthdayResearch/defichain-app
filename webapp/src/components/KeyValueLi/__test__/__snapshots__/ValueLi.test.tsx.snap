// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueLi component should check for snapshot 1`] = `
<Fragment>
  <a
    href="#"
    onClick={[Function]}
  >
    <div>
      <div>
        <div>
          <div
            className=""
          >
            Copied!
          </div>
          <EllipsisText
            length={50}
            tail="..."
            tailClassName="more"
            text="value"
          />
        </div>
      </div>
    </div>
  </a>
  <div />
</Fragment>
`;
